core/Columns/Dimension.php

```diff
@@ -85,16 +424,110 @@ public function hasImplementedEvent($method)
     }
 
     /**
-     * Adds a new segment. The segment type will be set to 'dimension' automatically if not already set.
+     * Adds a new segment. It automatically sets the SQL segment depending on the column name in case none is set
+     * already.
+     * @see \Piwik\Columns\Dimension::addSegment()
      * @param Segment $segment
      * @api
      */
     protected function addSegment(Segment $segment)
     {
-        $type = $segment->getType();
+        if (!$segment->getSegment() && $this->segmentName) {
```

Comment:
We now automatically enrich segment information when possible. Meaning a user can also easily add a segment just by doing this:

```php

        $segment = new Segment();
        $segment->setType(Segment::TYPE_METRIC);
        $this->addSegment($segment);
```